### PR TITLE
[Logigear] - Source Code.

### PR DESCRIPTION
**Pull request by NgoAnhKhoi**
Test case 3 & 4

### CODE REVIEW

**Comment by VladimirWork** [comment]
Looks like this code repeats itself in both test cases. Why don't we extend and use client.py class and pytest asserts to mark steps passed or failed (like in test_remove_add_roles.py)?

File: acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    test_report = TestReport("Test_scenario_03_Check_Connection")
+    test_results = {"Step 1": False, "Step 2": False, "Step 3": False, "Step 4": False,
+                    "Step 5": False, "Step 6": False, "Step 7": False, "Step 8": False}
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def test_precondition():
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py]: Looks like this code is used in all the tests. Can we create a common method for this?

File: acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+
+async def test_scenario_03_check_connection():
+    logger.info("Test scenario 3 -> started")
+
+    seed_steward01 = "000000000000000000000000Steward1"
+    pool_config = json.dumps({"genesis_txn": str(Constant.pool_genesis_txn_file)})
+
+    # 1. Create pool ledger
+    print(Colors.HEADER + "\n\t1.  Create pool ledger\n" + Colors.ENDC)
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py]: The same pattern/template is used for each test case (print/try/except).
Can we create an utility method and use this instead of boilerplate code?
I think it will increase test readability and maintenance a lot.

File: acceptance/indy_acceptance/indy-tests/utils/constant.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    TRUSTEE = "TRUSTEE"
+    STEWARD = "STEWARD"
+    TRUST_ANCHOR = "TRUST_ANCHOR"
+    TGB = "TGB"
+    NONE = ""
+
+
+class Constant:
+    import os
+    work_dir = os.path.expanduser('~') + os.sep + ".indy"
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/utils/constant.py]: The current working dir for libindy is `.indy_client`, not `.indy` (in the latest stable)

File: acceptance/indy_acceptance/indy-tests/utils/constant.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    seed_default_trustee = "000000000000000000000000Trustee1"
+    pool_genesis_txn_file = os.path.expanduser('~') + os.sep + ".sovrin/pool_transactions_sandbox_genesis"
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/utils/constant.py]: In the latest master and RC of indy-node, pool_genesis_txn_files are located in 
`/var/lib/indy/sandbox/pool_transactions_genesis` (for sandbox network)

File: acceptance/indy_acceptance/indy-tests/test_results/Acceptance_Test_20171117_17-42-58/Test_Scenario_09_Remove_And_Add_Role_20171117_17-42-58.json [hunk 5]
```diff
@@ -0,0 +1,173 @@
+{
+  "testcase": "Test_Scenario_09_Remove_And_Add_Role",
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/test_results/Acceptance_Test_20171117_17-42-58/Test_Scenario_09_Remove_And_Add_Role_20171117_17-42-58.json]: Is this file needed for tests?

File: acceptance/indy_acceptance/indy-tests/test_results/Acceptance_Test_20171117_17-42-58/Test_Scenario_09_Remove_And_Add_Role_20171117_17-42-58.log [hunk 6]
```diff
@@ -0,0 +1,118 @@
+[95m
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/test_results/Acceptance_Test_20171117_17-42-58/Test_Scenario_09_Remove_And_Add_Role_20171117_17-42-58.log]: It looks like a log. I think it needs to be removed

File: acceptance/indy_acceptance/indy-tests/utils/constant.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+
+class Constant:
+    """ 
+    Class Constant store some necessary paths.
+    """
+    import os
+    work_dir = os.path.expanduser('~') + os.sep + ".indy"
+    seed_default_trustee = "000000000000000000000000Trustee1"
+    pool_genesis_txn_file = os.path.expanduser('~') + os.sep + ".sovrin/pool_transactions_sandbox_genesis"
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/utils/constant.py]: In the latest master and RC of indy-node, pool_genesis_txn_files are located in
/var/lib/indy/sandbox/pool_transactions_genesis (for sandbox network)

File: acceptance/indy_acceptance/indy-tests/utils/constant.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    TGB = "TGB"  # obsolete
+    NONE = ""
+
+
+class Constant:
+    """ 
+    Class Constant store some necessary paths.
+    """
+    import os
+    work_dir = os.path.expanduser('~') + os.sep + ".indy"
```
Comment (ashcherbakov) [inline acceptance/indy_acceptance/indy-tests/utils/constant.py]: The current working dir for libindy is .indy_client, not .indy (in the latest stable)

Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: When we run tests with libindy, the .indy directory is created automatically.  I do not believe .indy_client is the default directory anymore.

File: acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py (hunk 1 above)
Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py]: I agree.  Khoi, please make it so

File: acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py (hunk 2 above)
Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/test_scenarios/test_scenario_03.py]: I agree. Khoi, please make it so

File: acceptance/indy_acceptance/indy-tests/utils/constant.py (hunk 4 above)
Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: This path is configurable so the user can use multiple transaction files in multiple locations

File: acceptance/indy_acceptance/indy-tests/utils/constant.py (hunk 8 above)
Comment (ashcherbakov) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: What version of libindy are you using?
I think it was changed in the latest stable (which was issued last week)

File: acceptance/indy_acceptance/indy-tests/utils/constant.py (hunk 4 above)
Comment (ashcherbakov) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: yes, but if you take config files from a sovrin installation, then they will be placed to the location mentioned above (and not to .sovrin)

Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: Khoi, please add a comment above this line that tells the user the path to the genesis transaction file is configurable.  The comment should indicate the default directory for the genesis transaction file is /var/lib/indy/sandbox/ but the user can put the genesis transaction file anywhere they want as long as the path is set in this line.  Please remove .sovrin/ and put the default path as /var/lib/indy/sandbox/ before checking this file in.

File: acceptance/indy_acceptance/indy-tests/utils/constant.py (hunk 8 above)
Comment (StevenL2015) [reply acceptance/indy_acceptance/indy-tests/utils/constant.py]: I am using the version from the stable build dated 01-Nov-2017.  That is the only build I see in the 1.0.1 directory
